From 78b250f7920353c2e41944b4710aade46958ed9e Mon Sep 17 00:00:00 2001
From: Dylan McCall <dylan@dylanmccall.ca>
Date: Thu, 4 Apr 2019 16:19:26 -0700
Subject: [PATCH] libdocument: Add transfer annotation for ev_document_get_info

The GIR annotation currently defaults to transfer-ownership="full",
which results in a double-free when using this function through GIR.
Change this to "transfer none" to let GIR know that it should not free
the returned EvDocumentInfo.

Closes #1125
---
 libdocument/ev-document.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libdocument/ev-document.c b/libdocument/ev-document.c
index d793fd25..9b2c323f 100644
--- a/libdocument/ev-document.c
+++ b/libdocument/ev-document.c
@@ -858,6 +858,14 @@ _ev_document_get_info (EvDocument *document)
 	return klass->get_info (document);
 }
 
+/**
+ * ev_document_get_info:
+ * @document: a #EvDocument
+ *
+ * Returns the #EvDocumentInfo for the document.
+ *
+ * Returns: (transfer none): a #EvDocumentInfo
+ */
 EvDocumentInfo *
 ev_document_get_info (EvDocument *document)
 {
-- 
2.18.1

